clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```diff
@@ -293,6 +294,16 @@ long getExpirationTimeForTimeout(final long timeoutMs) {
         return expiration;
     }
 
+    private <T> void chain(final CompletableFuture<T> a, final CompletableFuture<T> b) {
+        a.whenComplete((value, exception) -> {
+            if (exception != null) {
+                b.completeExceptionally(exception);
+            } else {
+                b.complete(value);
+            }
+        });
+    }
+
```

Comment:
`chain()` has been in `CompletedApplicationEvent` since its inception, but since that method is only used in this one class, I thought I'd migrate it here.